### PR TITLE
Dashboard/endpoint/fix

### PR DESCRIPTION
**Pull request by Gerhut**
- Transfer restfulapi error message when 403
- Use vanilla fetch to get the error message
- reset the switch when fails to enable

### CODE REVIEW

**Comment by Anbang-Hu** [comment]
I don't see the feature working properly in the cluster that has interactive limit deployed. I am unsure if which part goes wrong. @xudifsd can you help look into it?

**Comment by Anbang-Hu** [comment]
> I don't see the feature working properly in the cluster that has interactive limit deployed. I am unsure if which part goes wrong. @xudifsd can you help look into it?

Looks like I was in another security group that has admin access to the VC.

File: src/dashboard/src/pages/Job/Tabs/Endpoints.tsx [hunk 1]
```diff
@@ -112,15 +115,19 @@ const EndpointsController: FunctionComponent<{ endpoints: any[] }> = ({ endpoint
...
-  const { post } =
-    useFetch(`/api/clusters/${clusterId}/jobs/${jobId}/endpoints`,
-    [clusterId, jobId]);
   const portInput = useRef<HTMLInputElement>();
   const onChange = useCallback((name: string) => (event: ChangeEvent<{}>, value: boolean) => {
     if (value === false) return;
     enqueueSnackbar(`Enabling ${name}...`);
-    post({
-      endpoints: [name.toLowerCase()]
+    // TODO: revert to use-http after upgrade to 1.0
```
Comment (Anbang-Hu) [inline src/dashboard/src/pages/Job/Tabs/Endpoints.tsx:122]: Why revert after upgrade?

**Comment by Gerhut** [comment]
Backend continues on #1197 
Frontend continues on #1199

**Pull request closed without merging**